code/modules/reagents/chemistry_machinery/chem_master.dm

```diff
@@ -98,323 +98,298 @@
-				var/amount = text2num(href_list["amount"])
-				transfer_chemicals(src, beaker, amount, id)
-
-		else if(href_list["addcustom"])
-			var/id = href_list["addcustom"]
-			useramount = tgui_input_number(usr, "Select the amount to transfer.", "Transfer amount", useramount)
-			transfer_chemicals(src, beaker, useramount, id)
-
-		else if(href_list["addall"])
-			for(var/datum/reagent/R in beaker.reagents.reagent_list)
-				var/amount = beaker.volume
-				transfer_chemicals(src, beaker, amount, R.id)
-
-		else if(href_list["remove"])
-			if(href_list["amount"])
-				var/id = href_list["remove"]
-				var/amount = text2num(href_list["amount"])
-				if(mode)
-					transfer_chemicals(beaker, src, amount, id)
-				else
-					transfer_chemicals(null, src, amount, id)
+			loaded_pill_bottle.choose_color(user)
+
+		if("add")
+			var/amount = params["amount"]
+			var/id = params["id"]
+			if(!isnum(amount) || !id)
+				return
+
+			transfer_chemicals(src, beaker, amount, id)
+			return TRUE
 
+		if("add_all")
+			for(var/datum/reagent/beaker_reagent in beaker.reagents.reagent_list)
+				transfer_chemicals(src, beaker, beaker.volume, beaker_reagent.id)
+
+			return TRUE
+
+		if("remove")
+			var/amount = params["amount"]
+			var/id = params["id"]
+			if(!isnum(amount) || !id)
+				return
 
-		else if(href_list["removecustom"])
-			var/id = href_list["removecustom"]
-			useramount = tgui_input_number(usr, "Select the amount to transfer.", "Transfer amount", useramount)
 			if(mode)
-				transfer_chemicals(beaker, src, useramount, id)
-			else
-				transfer_chemicals(null, src, useramount, id)
+				transfer_chemicals(beaker, src, amount, id)
+				return TRUE
+
+			transfer_chemicals(null, src, amount, id)
+			return TRUE
 
-		else if(href_list["removeall"])
-			for(var/datum/reagent/R in src.reagents.reagent_list)
-				var/amount = src.reagents.total_volume
+		if("remove_all")
+			for(var/datum/reagent/contained_reagent in reagents.reagent_list)
+				var/amount = reagents.total_volume
 				if(mode)
-					transfer_chemicals(beaker, src, amount, R.id)
+					transfer_chemicals(beaker, src, amount, contained_reagent.id)
 				else
-					transfer_chemicals(null, src, amount, R.id)
+					transfer_chemicals(null, src, amount, contained_reagent.id)
 
-		else if(href_list["toggle"])
+			return TRUE
+
+		if("toggle")
 			mode = !mode
+			return TRUE
 
-		else if(href_list["main"])
-			attack_hand(user)
-			return
-		else if(href_list["eject"])
+		if("eject")
 			if(!beaker)
 				return
 
-			if(!Adjacent(usr) || !usr.put_in_hands(beaker))
+			if(!user.put_in_hands(beaker))
 				beaker.forceMove(loc)
 
 			beaker = null
 			reagents.clear_reagents()
 			update_icon()
+			return TRUE
 
-		else if (href_list["createpill"] || href_list["createpill_multiple"])
-			var/count = 1
+		if("create_pill")
+			if(!pill_maker)
+				return
 
-			if(reagents.total_volume/count < 1) //Sanity checking.
+			var/to_create = floor(clamp(params["number"], 1, max_pill_count))
```

Comment:
this could runtime if params["number"] is not a number
also clamp can take a list, which will then return a list so that's also something to watch out for

you'll wanna sanitize this param

Reply:
addressed ty